metadata/metadata.go

```diff
@@ -90,6 +90,22 @@ func Pairs(kv ...string) MD {
 	return md
 }
 
+// String implements the Stringer interface for pretty-printing a MD. Ordering of the values is non-deterministic as it ranges over a map.
+func (md MD) String() string {
+	var sb strings.Builder
+	fmt.Fprintf(&sb, "MD{")
+	needSep := false
+	for k, v := range md {
+		if needSep {
```

Comment:
Probably we could avoid this variable and just check for the base value("MD{")

Reply:
Done using the length of the builder.